ExampleMod/Common/UI/ResourceOverlay/VanillaLifeOverlay.cs

```diff
@@ -0,0 +1,140 @@
+				// Draw over the Fancy heart panels
+				DrawFancyPanelOverlay(context);
+			} else if (drawingBarsPanels) {
+				// Draw over the Bars middle life panels
+				DrawBarsPanelOverlay(context);
+			}
+		}
+
+		private bool CompareAssets(Asset<Texture2D> existingAsset, string compareAssetPath) {
+			// This is a helper method for checking if a certain vanilla asset was drawn
+			if (!vanillaAssetCache.TryGetValue(compareAssetPath, out var asset))
+				asset = vanillaAssetCache[compareAssetPath] = Main.Assets.Request<Texture2D>(compareAssetPath);
+
+			return existingAsset == asset;
+		}
+
+		private static void DrawClassicFancyOverlay(ResourceOverlayDrawContext context) {
+			// Draw over the Classic / Fancy hearts
+			// "context" contains information used to draw the resource
+			// If you want to draw directly on top of the vanilla hearts, just replace the texture and have the context draw the new texture
+			context.texture = ModContent.Request<Texture2D>("ExampleMod/Common/UI/ResourceOverlay/ClassicLifeOverlay", AssetRequestMode.ImmediateLoad);
+			context.Draw();
+		}
+
+		private static void DrawFancyPanelOverlay(ResourceOverlayDrawContext context) {
+			// Draw over the Fancy heart panels
+			string texture = "ExampleMod/Common/UI/ResourceOverlay/FancyLifeOverlay_Panel";
+			// The original position refers to the entire panel slice.
+			// However, since this overlay only modifies the "inner" portion of the slice (aka the part behind the heart),
+			// the position should be modified to compensate for the sprite size difference
+			Vector2 positionOffset;
+
+			if (context.resourceNumber == context.snapshot.AmountOfLifeHearts) {
+				// Final panel to draw has a special "Fancy" variant.  Determine whether it has panels to the left of it
+				if (context.resourceNumber % 10 == 1) {
+					// First and only panel in this panel's row
+					// Vanilla texture is "Heart_Single_Fancy"
+					positionOffset = new Vector2(8, 8);
+				} else {
+					// Other panels existed in this panel's row
+					// Vanilla texture is "Heart_Right_Fancy"
+					positionOffset = new Vector2(8, 8);
+				}
+			} else if (context.resourceNumber % 10 == 1) {
+				// First panel in this row
+				// Vanilla texture is "Heart_Left"
+				positionOffset = new Vector2(4, 4);
+			} else if (context.resourceNumber % 10 <= 9) {
+				// Any panel that has a panel to its left AND right
+				// Vanilla texture is "Heart_Middle"
+				positionOffset = new Vector2(0, 4);
+			} else {
+				// Final panel in the first row
+				// Vanilla texture is "Heart_Right"
+				positionOffset = new Vector2(0, 4);
+			}
+
+			// "context" contains information used to draw the resource
+			// If you want to draw directly on top of the vanilla hearts, just replace the texture and have the context draw the new texture
+			context.texture = ModContent.Request<Texture2D>(texture, AssetRequestMode.ImmediateLoad);
```

Comment:
1) fair point.  this is a leftover from the old code that had multiple textures
2) this line replaces the texture drawn, then draws the new texture, hence `=` instead of `??=`

Reply:
```cs
// field
Asset<Texture2D> panelTexture;

// in method later
context.texture = panelTexture ??= ModContent.Request<Texture2D>(texture);
```

Reply:
ah, that's what you meant